### ISSUE TITLE
Wif behaving differently when being used in Electron Application Production 

### ISSUE THREAD

**Issue by Taha-daboussi** [bitcoinjs/wif#33]
So am using "ecpair" library to get ECPairFactory and then using @bitcoinerlab/secp256k1 
as the ecc interface so I can import Bitcoin taproot private key.
When running this code in development it worked perfectly , but when I built it into my Electron application I ran into 1 problem 
getting an error of " expected buffer , received Uint8Array " 
after debugging the code  I found out where the issue was 

in the node_moules/wif/index.js function decodeRaw

![image](https://github.com/bitcoinjs/wif/assets/68201777/f2a24b9b-4590-40dd-b0d1-4236c6755cd9)

for some reason returning two different values if production or development . was able to fix this (not really a fix )

by doing this in the "decode" function 

```
function decode (string, version) {
  const data = decodeRaw(bs58check.decode(string), version) 
  if(data.privateKey instanceof Uint8Array ){
    const uint8Array = new Uint8Array(Object.keys(data.privateKey).map(key => data.privateKey[key]));
// Convert Uint8Array to Buffer
    const buffer = Buffer.from(uint8Array);
    data.privateKey = buffer
  }
  return data
}
```

Please let me know if there is any other solution other my temp fix

**Comment by junderw** [comment]
This sounds like an Electron issue. Treating Uint8Array as interchangeable with Buffer is probably not a good idea.

I assume their production settings cause something to get changed, or maybe it's the bundler's problem.

**Comment by junderw** [comment]
Check if fixed in v5.0.0
